service/src/main/java/bio/terra/cbas/dependencies/github/GitHubService.java

```diff
@@ -41,17 +57,50 @@ public String getCurrentGithash(
     return commitInfo.getSha();
   }
 
-  public String getBaseUrl(String organization, String repo, String branch, BearerToken userToken)
-      throws GitHubClient.GitHubClientException {
-    GitHubClient.CommitInfo commitInfo;
+  // If the methodVersion has methodVersionDetails, reconstruct the url using them
+  public static String getOrRebuildGithubUrl(MethodVersion methodVersion)
+      throws MethodProcessingException {
+    Optional<GithubMethodVersionDetails> methodVersionDetailsOptional =
+        methodVersion.methodVersionDetails();
+    Optional<GithubMethodDetails> methodDetailsOptional =
+        methodVersion.method().githubMethodDetails();
+
+    if (methodVersionDetailsOptional.isEmpty() || methodDetailsOptional.isEmpty()) {
+      return methodVersion.url();
+    } else {
+      GithubMethodDetails methodDetails = methodDetailsOptional.get();
+      GithubMethodVersionDetails methodVersionDetails = methodVersionDetailsOptional.get();
+      return buildRawGithubUrl(
+          methodDetails.organization(),
+          methodDetails.repository(),
+          methodVersionDetails.githash(),
+          methodDetails.path());
+    }
+  }
+
+  public static String buildRawGithubUrl(String org, String repo, String githash, String path) {
+    return "https://%s/%s/%s/%s/%s".formatted(RAW_GITHUB_URL_HOST, org, repo, githash, path);
+  }
+
+  // helper method to verify that URL is valid and its host is supported
+  public static String validateGithubUrl(String methodUrl) {
```

Comment:
when I look at how this method is used, it makes sense, but its name makes me think it should return a boolean and throw exceptions instead of returning a string or null. at the risk of nit-picking, maybe this should be called something like `generateGithubUrlErrorMessage`?